### PR TITLE
Fix error in the api doc about media and code HTTP 201 and 204

### PR DESCRIPTION
**Pull request by ahocquard**
This PR fix:
- HTTP code 201 and 204 used when patching a resource
- location header returned in 201 and 204 responses
- error message of the error 415 when creating media a media

### CODE REVIEW

File: content/akeneo-web-api.yaml [hunk 1]
```diff
@@ -372,6 +372,8 @@ paths:
           schema:
             $ref: "#/definitions/Product"
       responses:
+        201:
+          $ref: "#/responses/Created"
```
Comment (skeleton) [inline content/akeneo-web-api.yaml]: remove 404 please on each patch action

File: content/akeneo-web-api.yaml [hunk 2]
```diff
@@ -2462,6 +2476,15 @@ responses:
       "code": 415,
       "message": "‘xxx’ in ‘Content-type’ header is not valid.  Only ‘application/vnd.akeneo.collection+json’ is allowed."
     }
+  415ErrorMultipart:
+    description: Unsupported Media type
+    x-details: The `Content-type` header is not `multipart/form-data` but it should
```
Comment (skeleton) [inline content/akeneo-web-api.yaml]: The `Content-type` header has to be `multipart/form-data`
?

Comment (ahocquard) [inline content/akeneo-web-api.yaml]: I've just modified `application/json` into `multipart/form-data`. The other sentences (for `application/json` and `application/vnd.akeneo.collection+json` are the same. But I can modify it (both sentences are ok for me).

Comment (skeleton) [inline content/akeneo-web-api.yaml]: ok, I found the first sentence very weird, but you know my english level ;)

Comment (ahocquard) [inline content/akeneo-web-api.yaml]: I've modified it, it seems better

**Pull request merged**